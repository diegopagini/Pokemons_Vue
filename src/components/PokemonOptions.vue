<!-- @format -->

<template>
	<h3>Pokemon options</h3>
</template>

<script>
	export default {};
</script>
