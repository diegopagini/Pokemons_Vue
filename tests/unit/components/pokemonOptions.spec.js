/** @format */
import PokemonOptions from '@/components/PokemonOptions';
import { shallowMount } from '@vue/test-utils';

import { POKEMONS_MOCK } from '../mocks/pokemons';

describe('PokemonOptions Component', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallowMount(PokemonOptions, {
			props: {
				options: POKEMONS_MOCK,
			},
		});
	});

	test('debe de hacer match con el snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	test('debe de mostrar las 4 opciones correctamente', () => {
		const elements = wrapper.findAll('li');
		expect(elements.length).toBe(4);
		expect(elements.at(0).text()).toBe(POKEMONS_MOCK[0].name);
		elements.forEach((el) => {
			expect(el.text()).toEqual(expect.any(String));
		});
	});

	test('debe de emitir "selection" cons sus respectivos parÃ¡metros al hacer click', () => {
		const [l1, l2, l3, l4] = wrapper.findAll('li');

		l1.trigger('click');
		l2.trigger('click');
		l3.trigger('click');
		l4.trigger('click');

		expect(wrapper.emitted('selection').length).toBe(4);
		expect(wrapper.emitted('selection').at(0)).toEqual([1]);
		expect(wrapper.emitted('selection').at(1)).toEqual([2]);
		expect(wrapper.emitted('selection').at(2)).toEqual([3]);
		expect(wrapper.emitted('selection').at(3)).toEqual([4]);
	});
});
