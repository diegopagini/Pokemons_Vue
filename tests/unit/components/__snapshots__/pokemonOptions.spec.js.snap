// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul class="fade-in">
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
